antenna/app.py

```diff
@@ -144,107 +50,114 @@ def build_config_manager():
-    )
-    required_config.add_option(
-        "secret_sentry_dsn",
-        default="",
-        doc=(
-            "Sentry DSN to use. See https://docs.sentry.io/quickstart/#configure-the-dsn "
-            "for details. If this is not set an unhandled exception logging middleware "
-            "will be used instead."
-        ),
-    )
-    required_config.add_option(
-        "host_id",
-        default="",
-        doc=(
-            "Identifier for the host that is running Antenna. This identifies this Antenna "
-            "instance in the logs and makes it easier to correlate Antenna logs with "
-            "other data. For example, the value could be a public hostname, an instance id, "
-            "or something like that. If you do not set this, then socket.gethostname() is "
-            "used instead."
-        ),
-    )
+    class Config:
+        basedir = Option(
+            default=str(Path(__file__).parent.parent),
+            doc="The root directory for this application to find and store things.",
+        )
+        logging_level = Option(
+            default="DEBUG",
+            doc="The logging level to use. DEBUG, INFO, WARNING, ERROR or CRITICAL",
+        )
+        local_dev_env = Option(
+            default="False",
+            parser=bool,
+            doc="Whether or not this is a local development environment.",
+        )
+        statsd_host = Option(default="localhost", doc="Hostname for statsd server.")
+        statsd_port = Option(default="8125", doc="Port for statsd server.", parser=int)
+        statsd_namespace = Option(default="", doc="Namespace for statsd metrics.")
+        secret_sentry_dsn = Option(
+            default="",
+            doc=(
+                "Sentry DSN to use. See https://docs.sentry.io/quickstart/#configure-the-dsn "
+                "for details. If this is not set an unhandled exception logging middleware "
+                "will be used instead."
+            ),
+        )
+        host_id = Option(
+            default="",
+            doc=(
+                "Identifier for the host that is running Antenna. This identifies this Antenna "
+                "instance in the logs and makes it easier to correlate Antenna logs with "
+                "other data. For example, the value could be a public hostname, an instance id, "
+                "or something like that. If you do not set this, then socket.gethostname() is "
+                "used instead."
+            ),
+        )
 
     def __init__(self, config):
+        super().__init__()
         self.config_manager = config
         self.config = config.with_options(self)
+        self._all_resources = {}
 
-    def __call__(self, key):
-        """Return configuration for given key."""
-        return self.config(key)
+        # This is the crashmover which takes a crash report, saves it, and publishes
+        # it
+        self.crashmover = CrashMover(config.with_namespace("crashmover"))
```

Comment:
I extracted the crashmover code from `BreakpadSubmitterResource` so if we ever add support for other payload formats, they can also use the crashmover.